src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -219,7 +229,7 @@ protected ClusterTopologyInfo queryForTopology(JdbcConnection conn) throws SQLEx
       // "Unknown table 'REPLICA_HOST_STATUS' in information_schema"
       // Ignore this kind of exceptions.
     } finally {
-      if (this.gatherPerfMetrics) {
+      if (topologyInfo != null && conn.getPropertySet() != null && conn.getPropertySet().getBooleanProperty(PropertyKey.gatherPerfMetrics.getKeyName()).getValue()) {
```

Comment:
You need to extract this condition into a function

Reply:
We need to check if that "gatherPerfMetrics" parameter is still available after failover on a new connection.